### PR TITLE
Feature/client restaurant/add meal

### PR DESCRIPTION
**Pull request by NadaFeteiha**
The image selection is not complete yet, and validation for the name ..etc, are same

### CODE REVIEW

File: client_restaurant/shared/src/commonMain/kotlin/presentation/meal/MealScreenInteractionListener.kt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package presentation.meal
+
+import presentation.base.BaseInteractionListener
+
+interface MealScreenInteractionListener : BaseInteractionListener {
+    fun onImageClicked()
+    fun onNameChanged(name: String)
+    fun onDescriptionChanged(description: String)
+    fun onPriceChanged(price: String)
```
Comment (Saif-Alhaider) [inline client_restaurant/shared/src/commonMain/kotlin/presentation/meal/MealScreenInteractionListener.kt:9]:
```suggestion
    fun onImageClick()
    fun onNameChange(name: String)
    fun onDescriptionChange(description: String)
    fun onPriceChange(price: String)
```

File: client_restaurant/shared/src/commonMain/kotlin/presentation/meal/MealScreenUIState.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    val cuisines: List<CuisineUIState> = emptyList()
+)
+
+
+data class CuisineUIState(
+    val id: String = "",
+    val name: String = ""
+)
+
+fun List<CuisineUIState>.toCuisinesString() = this.joinToString(", ") { it.name }
```
Comment (Saif-Alhaider) [inline client_restaurant/shared/src/commonMain/kotlin/presentation/meal/MealScreenUIState.kt:18]:
```suggestion
fun List<CuisineUIState>.toCuisinesString() =
    joinToString(", ", transform = CuisineUIState::name)
```

File: client_restaurant/shared/src/commonMain/kotlin/resources/StringResources.kt [hunk 3]
```diff
@@ -2,5 +2,11 @@ package resources
 
 
 data class StringResources(
-    val beepBeep : String = "Beep Beep"
+    val beepBeep: String = "Beep Beep",
```
Comment (Saif-Alhaider) [inline client_restaurant/shared/src/commonMain/kotlin/resources/StringResources.kt:5]: when are you using beepbeep ?

File: client_restaurant/shared/src/commonMain/kotlin/resources/StringResources.kt [hunk 4]
```diff
@@ -2,5 +2,11 @@ package resources
...
+    val buttonAdd: String = "Add",
```
Comment (Saif-Alhaider) [inline client_restaurant/shared/src/commonMain/kotlin/resources/StringResources.kt:6]: change variable name to add

**Review by Saif-Alhaider — COMMENTED** [review]
requires some refactor

**Review by HaidyAbuGom3a — APPROVED** [review]

**Review by mosayed01 — APPROVED** [review]

**Review by ahmednasserzaza — APPROVED** [review]

**Pull request merged**